fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -317,6 +323,114 @@ private void updateDimensionRow(
+
+            LOG.debug(
+                    "Deleting {} since new Lucene indexes have been moved away from there and is now empty",
+                    newLuceneIndexPathString
+            );
+            deleteDir(newLuceneIndexPathString);
+
+            LOG.debug("Deleting old Lucene indexes in {} ...", tempDir);
+            deleteDir(tempDir);
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    /**
+     * Moves all files and sub-directories from one location to another.
+     * <p>
+     * Two locations must exist before calling this method.
+     *
+     * @param sourceDir  The location where files and sub-directories will be moved from
+     * @param destinationDir  The location where files and sub-directories will be moved to
+     */
+    private static void moveDirEntries(String sourceDir, String destinationDir) {
+        Path sourcePath = Paths.get(sourceDir).toAbsolutePath();
+        Path destinationPath = Paths.get(destinationDir).toAbsolutePath();
+
+        if (!Files.exists(destinationPath)) {
+            try {
+                Files.createDirectory(destinationPath);
+            } catch (IOException e) {
+                LOG.error(ErrorMessageFormat.CANNOT_CREATE_INDEX_DIR.format(destinationDir));
+                throw new RuntimeException(e);
+            }
+        }
+
+        try {
+            Files.walkFileTree(sourcePath, new SimpleFileVisitor<Path>() {
+                @Override
+                public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes basicFileAttributes)
+                        throws  IOException {
+                    Path destinationDirPath = destinationPath.resolve(sourcePath.relativize(dir));
+                    if (!Files.exists(destinationDirPath)) {
+                        Files.createDirectory(destinationDirPath);
+                        LOG.trace("Creating sub-directory {} under {} ...", dir, destinationDir);
+                    }
+                    return FileVisitResult.CONTINUE;
+                }
+
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes basicFileAttributes)
+                        throws IOException {
+                    Path destinationFileName = destinationPath.resolve(sourcePath.relativize(file));
+                    LOG.trace("Moving {} to {}", file, destinationFileName);
+                    Files.move(file, destinationFileName);
+                    return FileVisitResult.CONTINUE;
+                }
+            });
+        } catch (IOException e) {
+            LOG.error("I/O error thrown by SimpleFileVisitor method");
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Deletes a directory and all entries under that directory.
+     *
+     * @param path  The location of the directory that is to be deleted
+     */
+    private static void deleteDir(String path) {
```

Comment:
There is a deleteFiles in Utils class......

Reply:
I forgot this works for deleting. Thanks!